static/app/components/dropdownMenuItemV2.tsx

```diff
@@ -105,137 +111,160 @@ type Props = {
+      const {menuItemProps, labelProps, descriptionProps} = useMenuItem(
+        {
+          key: node.key,
+          onAction: actionHandler,
+          onClose,
+          closeOnSelect,
+          isDisabled,
+        },
+        state,
+        ref
+      );
 
-    return (
-      <MenuItemWrap
-        ref={ref}
-        as={renderAs}
-        isDisabled={isDisabled}
-        {...props}
-        {...(isSubmenuTrigger && {role: 'menuitemradio'})}
-      >
-        <InnerWrap isFocused={isFocused} role="presentation">
-          {leadingItems && (
-            <LeadingItems
-              isDisabled={isDisabled}
-              spanFullHeight={leadingItemsSpanFullHeight}
-            >
-              {leadingItems}
-            </LeadingItems>
-          )}
-          <ContentWrap
-            isFocused={isFocused}
-            showDividers={showDividers}
-            role="presentation"
-          >
-            <LabelWrap role="presentation">
-              <Label isDisabled={isDisabled} {...labelProps} aria-hidden="true">
-                {label}
-              </Label>
-              {details && <Details {...descriptionProps}>{details}</Details>}
-            </LabelWrap>
-            {(trailingItems || isSubmenuTrigger) && (
-              <TrailingItems
+      // Merged menu item props, class names are combined, event handlers chained,
+      // etc. See: https://react-spectrum.adobe.com/react-aria/mergeProps.html
+      const props = mergeProps(
+        submenuTriggerProps,
+        menuItemProps,
+        hoverProps,
+        keyboardProps
+      );
+      const {
+        details,
+        leadingItems,
+        leadingItemsSpanFullHeight,
+        trailingItems,
+        trailingItemsSpanFullHeight,
+      } = item;
+      const label = node.rendered ?? item.label;
+      const showDividers = item.showDividers && !isLastNode;
+
+      return (
+        <MenuItemWrap
+          ref={ref}
+          as={renderAs}
+          isDisabled={isDisabled}
+          {...props}
+          {...(isSubmenuTrigger && {role: 'menuitemradio'})}
+        >
+          <InnerWrap isFocused={isFocused} role="presentation">
+            {leadingItems && (
+              <LeadingItems
                 isDisabled={isDisabled}
-                spanFullHeight={trailingItemsSpanFullHeight}
+                spanFullHeight={leadingItemsSpanFullHeight}
               >
-                {trailingItems}
-                {isSubmenuTrigger && (
-                  <IconChevron size="xs" direction="right" aria-hidden="true" />
-                )}
-              </TrailingItems>
+                {leadingItems}
+              </LeadingItems>
             )}
-          </ContentWrap>
-        </InnerWrap>
-      </MenuItemWrap>
-    );
-  }
+            <ContentWrap
+              isFocused={isFocused}
+              showDividers={showDividers}
+              role="presentation"
+            >
+              <LabelWrap role="presentation">
```

Comment:
is this correct? both shall have a role presentation?

Reply:
added these to simplify what's screen readers announce for each item (instead of saying `menu item group`, they should preferably say `menu item`). but it seems that react-aria takes care of that already! will delete these.